cmd/soroban-rpc/internal/methods/util.go

```diff
@@ -0,0 +1,53 @@
+package methods
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+func getProtocolVersion(
+	ctx context.Context,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) (uint32, error) {
+	var protocolVersion uint32
+	readTx, err := ledgerEntryReader.NewCachedTx(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		_ = readTx.Done()
+	}()
+
+	latestLedger, err := readTx.GetLatestLedgerSequence()
+	if err != nil {
+		return 0, err
+	}
+
+	_, protocolVersion, err = getBucketListSizeAndProtocolVersion(ctx, ledgerReader, latestLedger)
+	if err != nil {
+		return 0, err
+	}
+	return protocolVersion, nil
+}
+
+func getBucketListSizeAndProtocolVersion(
```

Comment:
This is too specific to have as a generic util function (that's why it was in `simulate_transaction.go`). The function is tiny, so it's fine to replicate the needed parts of `getBucketListSizeAndProtocolVersion()` into `getBucketListSizeAndProtocolVersion()` and leave it where it was.

This will also allow you to get the ledger sequence and ledger in the same transaction.

Reply:
Got it. replicated some part of it, but keeping `getProtocolVersion` in util